ui/app/routes/vault/cluster/logout.js

```diff
@@ -27,15 +27,17 @@ export default Route.extend(ModelBoundaryRoute, {
     this.console.clearLog(true);
     this.flashMessages.clearMessages();
     this.permissions.reset();
+
+    queryParams.with = authType;
+    if (ns) {
+      queryParams.namespace = ns;
+    }
     if (Ember.testing) {
       // Don't redirect on the test
-      this.replaceWith('vault.cluster.auth', { queryParams: { with: authType } });
+      this.replaceWith('vault.cluster.auth', { queryParams });
     } else {
-      let params = `?with=${authType}`;
```

Comment:
Looking at the `auth-form` component, if `wrappedToken` is provided it will automatically attempt to unseal the token and if successful it preforms the same action as if the form has been filled out and the user hits submit manually. If a namespace is present in the query param it should behave as if the user selected one in the dropdown.